src/cluster.c

```diff
@@ -2855,15 +2910,14 @@ void clusterBroadcastMessage(void *buf, size_t len) {
         if (!node->link) continue;
         if (node->flags & (CLUSTER_NODE_MYSELF|CLUSTER_NODE_HANDSHAKE))
             continue;
-        clusterSendMessage(node->link,buf,len);
+        clusterSendMessage(node->link,msgblock);
     }
     dictReleaseIterator(di);
 }
 
 /* Build the message header. hdr must point to a buffer at least
  * sizeof(clusterMsg) in bytes. */
-void clusterBuildMessageHdr(clusterMsg *hdr, int type) {
-    int totlen = 0;
+void clusterBuildMessageHdr(clusterMsg *hdr, int type, size_t msglen) {
```

Comment:
I think in all cases we want this to be:
```suggestion
void clusterBuildMessageHdr(clusterMsgSendBlock *msg, int type) {
```

Reply:
Is the benefit of this moving `clusterMsg *hdr = &msgblock->msg;` inside clusterBuildMessageHdr?

Reply:
Basically es, since you do it quite frequently.

Reply:
One thing is that there is extra header building work being done afterwards in the calling code for several cases (clusterSendUpdate, clusterSendPing,...), so I'd want to have `clusterMsg *hdr = &msgblock->msg;` in that code too

Reply:
That is true, the other reason is because the msg has the built in length.

Reply:
Moving `clusterBuildMessageHdr` into `createClusterMsgSendBlock` to reduce duplication